cpp/src/arrow/compute/cast.cc

```diff
@@ -96,8 +96,11 @@ class CastMetaFunction : public MetaFunction {
     ARROW_ASSIGN_OR_RAISE(auto cast_options, ValidateOptions(options));
     // args[0].type() could be a nullptr so check for that before
     // we do anything with it.
+    // TODO: if types are equal except for field names of list types, we can
```

Comment:
Do you want to open a JIRA for this TODO?

Reply:
I'll be doing this in https://github.com/apache/arrow/pull/13851